### PR TITLE
Add Option to Return Actual Caught Error Message and Stack Trace when RPC Call Fails

### PR DESCRIPTION
**Pull request by scorbajio**
This change catches errors that may happen in RPC calls and returns them as well as a stack trace if the `rpcDebug` option is set to `true`. Also see #3044.

### CODE REVIEW

File: packages/client/src/rpc/modules/eth.ts [hunk 1]
```diff
@@ -385,10 +387,12 @@ export class Eth {
       const { execResult } = await vm.evm.runCall(runCallOpts)
       return bytesToHex(execResult.returnValue)
     } catch (error: any) {
-      throw {
-        code: INTERNAL_ERROR,
-        message: error.message.toString(),
+      const e: any = {
```
Comment (acolytec3) [inline packages/client/src/rpc/modules/eth.ts:390]: This all looks great.  How hard is it to add a type to the `e` value that's thrown by each handler?

**Review by acolytec3 — COMMENTED** [review]
This is looking good overall.  Did you explore any alternatives that don't require repeating essentially the same try/catch in every function call?  Would it be possible to put a new `callWithStackTrace` method in each class and then have it wrapp the underlying RPC method handlers in a try/catch and do that for every function call?  I guess the way I'm thinking about it, you'd have `eth.callWithStackTrace(this.getBlockByHash(...` or something like that.  And in the body of it, you just do: 
```
callWithStackTrace = (handler: async () => {}, debug: boolean) {
try {
  await   handler()
}
catch (err) {
  // Throw the error with or without stack trace
}
}
```
{

**Comment by jochem-brouwer** [comment]
@acolytec3 I was thinking the same, I think this is possible in the middleware? If you catch it there you could generally report stack traces (I think)

File: packages/client/src/rpc/modules/eth.ts (hunk 1 above)
Comment (scorbajio) [reply packages/client/src/rpc/modules/eth.ts]: Fixed

**Review by jochem-brouwer — COMMENTED** [review]
The changes in the methods of debug/eth (besides constructor) are there because now the `callWithStackTrace` handles the errors (if `INTERNAL_ERROR`) right?

Also, the CI of client fails :thinking: 

In general LGTM but I think something goes wrong with error handling since client CI fails (have not looked in detail)

**Review by acolytec3 — APPROVED** [review]
LGTM.  Tests should all pass now.

**Pull request merged**